Compiler.cmake

```diff
@@ -11,6 +11,7 @@
 # Input Variables
 # * COMMON_MINIMUM_GCC_VERSION check for a minimum gcc version, default 4.4
 # * COMMON_USE_CXX03 When set, do not enable C++11 language features
+# * COMMON_DISABLE When set, do not set any flags
```

Comment:
@juanchopanza: This file is the attempt for an 'bbp-official-cpp-flags cmake module'. So far the uptake has been medium, therefore I expect some haggling from you (and others).

We do not check for compatibility in flags yet. Not sure how this can be done.